v2/yarpcgrpc/integration_test.go

```diff
@@ -21,521 +21,391 @@
-			serverValidity:      time.Minute,
-			expectedErrContains: "remote error: tls: bad certificate",
-			name:                "invalid client cert",
+			name:                 "larger than client send",
+			value:                strings.Repeat("a", 10),
+			ClientMaxSendMsgSize: 1,
+			errCode:              yarpcerror.CodeResourceExhausted,
+		},
+		{
+			name:                 "larger than server send",
+			value:                strings.Repeat("a", 10),
+			ServerMaxSendMsgSize: 1,
+			errCode:              yarpcerror.CodeResourceExhausted,
+		},
+		{
+			name:                 "larger than client recieve",
+			value:                strings.Repeat("a", 10),
+			ClientMaxRecvMsgSize: 1,
+			errCode:              yarpcerror.CodeResourceExhausted,
 		},
 	}
 
-	for _, test := range tests {
-		t.Run(test.name, func(t *testing.T) {
-			scenario := createTLSScenario(t, test.clientValidity, test.serverValidity)
-
-			serverCreds := credentials.NewTLS(&tls.Config{
-				GetCertificate: func(_ *tls.ClientHelloInfo) (*tls.Certificate, error) {
-					return &tls.Certificate{
-						Certificate: [][]byte{scenario.ServerCert.Raw},
-						Leaf:        scenario.ServerCert,
-						PrivateKey:  scenario.ServerKey,
-					}, nil
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			te := testEnvOptions{
+				Procedures: yarpcjson.Procedure("test-procedure", testEchoHandler),
+				Inbound: &Inbound{
+					ServerMaxRecvMsgSize: tt.ServerMaxRecvMsgSize,
+					ServerMaxSendMsgSize: tt.ServerMaxSendMsgSize,
 				},
-				ClientAuth: tls.RequireAndVerifyClientCert,
-				ClientCAs:  scenario.CAs,
-			})
-
-			clientCreds := credentials.NewTLS(&tls.Config{
-				GetClientCertificate: func(_ *tls.CertificateRequestInfo) (*tls.Certificate, error) {
-					return &tls.Certificate{
-						Certificate: [][]byte{scenario.ClientCert.Raw},
-						Leaf:        scenario.ClientCert,
-						PrivateKey:  scenario.ClientKey,
-					}, nil
+				Dialer: &Dialer{
+					ClientMaxRecvMsgSize: tt.ClientMaxRecvMsgSize,
+					ClientMaxSendMsgSize: tt.ClientMaxSendMsgSize,
 				},
-				RootCAs: scenario.CAs,
-			})
-
-			te := testEnvOptions{
-				InboundOptions: []InboundOption{InboundCredentials(serverCreds)},
-				DialOptions:    []DialOption{DialerCredentials(clientCreds)},
 			}
-			te.do(t, func(t *testing.T, e *testEnv) {
-				err := e.SetValueYARPC(context.Background(), "foo", "bar")
-				expectErrorContains(t, err, test.expectedErrContains)
 
-				err = e.SetValueGRPC(context.Background(), "foo", "bar")
-				expectErrorContains(t, err, test.expectedErrContains)
+			doWithTestEnv(t, te, func(t *testing.T, testEnv *testEnv) {
+				client := yarpcjson.New(testEnv.Client)
+				ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+				defer cancel()
+
+				var res testEchoResponse
+				err := client.Call(ctx, "test-procedure", &testEchoRequest{Message: tt.value}, &res)
+				require.Equal(t, tt.errCode.String(), yarpcerror.FromError(err).Code().String())
```

Comment:
Seeing as we are calling `String()` on both the expected and actual results, can we drop it?

Reply:
this was intentional actually, because if we have a failing test, we can see the string representations of the code, instead of an arbitrary number